clients/fides-js/src/components/tcf/TcfOverlay.tsx

```diff
@@ -246,14 +312,14 @@ const TcfOverlay: FunctionComponent<OverlayProps> = ({
         userLocationString: fidesRegionString,
         cookie,
         debug: options.debug,
-        servedNotices: null, // TODO: served notices
+        servedNotices: null,
```

Comment:
I think the thing to do here is to actually refactor NoticeOverlay a little too... I know that's not what this ticket calls for, but I think it'd be a good thing to do if you're up for it. basically, before TCF ever came along, we would:
1. In `NoticeOverlay` get the `servedNotices`
2. When the user saves, pass the `servedNotices` to `updateConsentPreferences` along with the `consentPreferencesToSave`
3. `updateConsentPreferences` would then do the unenviable task of [mapping `servedNotices` to `consentPreferencesToSave`](https://github.com/ethyca/fides/blob/main/clients/fides-js/src/lib/preferences.ts/#L54-L60)

Now that we have TCF, we:
1. In `TcfOverlay` get the `servedNotices`
2. `TcfOverlay` does the unenviable task of mapping `servedNotices` to each of the tcf entities (what your `noticeSubMap` is doing
3. We pass the `tcf` obj directly to `updateConsentPreferences` and set `servedNotices=null`

This is a good approach, and I think we should do the same for notices. In other words, I think `updateConsentPreferences` should no longer take a `servedNotices`, and we should pass an object that already has the `servedNotices` matched into `updateConsentPreferences`, same as we do for TCF. so then we can remove some code from `preferences.ts`. this might involve updating the type that `updateConsentPreferences` takes but hopefully it's not arduous. what do you think?

Reply:
Sounds good! I gave it a shot, let me know if that's what you had in mind.

Reply:
nice job, I think you handled it perfectly! 💯